services/common/src/components/comments/CommentEditor.tsx

```diff
@@ -0,0 +1,75 @@
+import React, { FC } from "react";
+import { useSelector } from "react-redux";
+import { Button, Form, Input } from "antd";
+import { getUserAccessData } from "@mds/common/redux/selectors/authenticationSelectors";
+import { USER_ROLES } from "@mds/common/index";
+
+interface CommentEditorProps {
+  onSubmit: (data: any) => void;
+  addCommentPermission: string;
+  maxLength?: number;
+}
+
+export const CommentEditor: FC<CommentEditorProps> = ({
+  onSubmit,
+  addCommentPermission,
+  maxLength = 300,
+}) => {
+  const userRoles = useSelector(getUserAccessData);
+
+  const [submitting, setSubmitting] = React.useState(false);
+  const [comment, setComment] = React.useState("");
+
+  const handleReset = () => {
+    setComment("");
+    setSubmitting(false);
+  };
+
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+    setSubmitting(true);
+
+    await onSubmit({ comment });
+    handleReset();
+  };
+
+  const handleChange = (e) => {
+    setComment(e.target.value);
+  };
+
+  const canAddComment = addCommentPermission
```

Comment:
I still think it would be more flexible to use userHasRole! Then you're not stuck using the Permission file at any point. just canAddComment = useSelector(userHasRole(addCommentPermission)) and then you can take out a bunch of imports too.